Yank/repex.py

```diff
@@ -1,752 +1,1428 @@
+                                                  "between states i and j from iteration 'iteration-1'."))
+
+        # Store statistics.
+        self._storage.variables['accepted'][iteration, :, :] = n_accepted_matrix[:, :]
+        self._storage.variables['proposed'][iteration, :, :] = n_proposed_matrix[:, :]
+
+    def read_timestamp(self, iteration):
+        """Return the timestamp for the given iteration.
 
-        # Clean up if we created a Context and Integrator.
-        if cleanup_context:
-            del context, integrator
+        Parameters
+        ----------
+        iteration : int
+            The iteration at which to read the data.
 
-        return potential_energy
+        Returns
+        -------
+        timestamp : str
+            The timestamp at which the iteration was stored.
 
-    def is_compatible_with(self, state):
         """
-        Determine whether another state is in the same thermodynamic ensemble (e.g. NVT, NPT).
+        return self._storage.variables['timestamp'][iteration]
 
-        ARGUMENTS
+    def write_timestamp(self, iteration):
+        """Store a timestamp for the given iteration.
 
-        state (ThermodynamicState) - thermodynamic state whose compatibility is to be determined
+        Parameters
+        ----------
+        iteration : int
+            The iteration at which to read the data.
+
+        """
+        # Create variable if needed.
+        if 'timestamp' not in self._storage.variables:
+            self._storage.createVariable('timestamp', str, ('iteration',), zlib=False, chunksizes=(1,))
+        self._storage.variables['timestamp'][iteration] = time.ctime()
 
-        RETURNS
+    def read_dict(self, name):
+        """Restore a dictionary from the storage file.
 
-        is_compatible (boolean) - True if 'state' is of the same ensemble (e.g. both NVT, both NPT), False otherwise
+        Parameters
+        ----------
+        name : str
+            The identifier of the dictionary used to stored the data.
 
-        EXAMPLES
+        Returns
+        -------
+        data : dict
+            The restored data as a dict.
 
-        Create NVT and NPT states.
+        """
+        ncgrp = self._storage.groups[name]
+        data = self._read_dict(ncgrp)
 
-        >>> from simtk import unit
-        >>> from openmmtools import testsystems
-        >>> testsystem = testsystems.LennardJonesCluster()
-        >>> [system, positions] = [testsystem.system, testsystem.positions]
-        >>> nvt_state = ThermodynamicState(system=system, temperature=100.0*unit.kelvin)
-        >>> npt_state = ThermodynamicState(system=system, temperature=100.0*unit.kelvin, pressure=1.0*unit.atmospheres)
+        # Restore the title in the metadata.
+        if name == 'metadata':
+            data['title'] = self._storage.title
+        return data
 
-        Test compatibility.
+    def write_dict(self, name, data):
```

Comment:
I'll patch this with your custom `YamlDumper/Loader` in a separate PR so that we can generate data that we'll be able to read later.